text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,799 @@
+  - Entity merging will occur first resulting in an "Entity Merged" Resource (See [algorithm here](#entity-merging-and-resource)).
+  - Resource detectors otherwise follow existing merge semantics.
+    - The Specification merge rules will be updated to account for violations prevalent in ALL implementation of resource detection.
+    - Specifically: This means the [rules around merging Resource across schema-url will be dropped](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#merge).  Instead only conflicting attributes will be dropped.
+    - SchemaURL on Resource will need to be deprecated with entity-specific schema-url replacing it. SDKs will no longer fill out SchemaURL on Resource. Additionally, as no (non-service) Resource semantic conventions have ever stabilized, SchemaURL usage on Resource cannot be in stable components of OpenTelemetry.  Given prevalent concerns of implementations around Resource merge specification, we suspect impact of this deprecation to be minimal.
+  - An OOTB ["Env Variable Entity Detector"](#environment-variable-detector) will be specified and provided vs. requiring SDK wide ENV variables for resource detection.
+- *Additionally, Resource Coordinator would be responsible for understanding Entity lifecycle events, for Entities whose lifetimes do not match or exceed the SDK's own lifetime (e.g. browser session).*
+
+#### Entity Detector
+
+The Entity detector in the SDK is responsible for detecting possible entities that could identify the SDK.  For Example, if the SDK is running in a kubernetes pod, it may provide an Entity for that pod.   SDK Entity Detectors are only required to provide identifying attributes, but may provide descriptive attributes to ensure combined Resource contains similar attributes as today's SDK.
+
+An Entity Detector would have an API similar to:
+
+```rust
+trait EntityDetector
+  pub fn detect_entities(...) -> Result<Vec<Entity>, EntityDetectionError>>
+```
+
+Where `Result` is the equivalent of error channel in the language of choice (e.g. in Go this would be `entities, err := e.detectEntities()`).
+
+#### Entity Merging and Resource
+
+The most important aspect of this design is how Entities will be merged to construct a Resource.
+
+We provide a simple algorithm for this behavior:
+
+- Construct a set of detected entities, `E`
+  - All entity detectors are sorted by priority (highest first)
+  - For each entity detector `D`, detect entities
+    - For each entity detected, `d'`
+      - If an entity `e'` exists in `E` with same entity type as `d'`, do one of the following:
+        - If the entity identiy and schema_url are the same, merge the descriptive attributes of `d'` into `e'`:
+          - For each descriptive attribute `da'` in `d'`
+            - If `da'.key` does not exist in `e'`, then add `da'` to `ei`
+            - otherwise, ignore.
```

Comment:
implementations should have an option to debug this case; in most cases, this should be an error

Reply:
Definitely disagree. In multi-observer case or even the use case of AWS Ec2 providing host entity, it's not guaranteed every observer *can* see all possible attributes.

Reply:
Ok, but then you disagree with the *most* cases part of my statement, right? I would hope to be able to design our observability in a way that every entity has a defined point of enrichment on the path between source and first store.

Reply:
yes "most" is what I disagree with.

Your goal for how you want to design o11y -> I think this is an ideal we should aim for, but we I don't think it's something we can force.

Reply:
That's why I requested to have an option ;)

Reply:
The design leaves available the option.

I think every scenario in this manager eneds to be debuggable.